yb-voyager/src/queryparser/query_parser.go

```diff
@@ -42,3 +45,45 @@ func ParsePLPGSQLToJson(query string) (string, error) {
 func GetProtoMessageFromParseTree(parseTree *pg_query.ParseResult) protoreflect.Message {
 	return parseTree.Stmts[0].Stmt.ProtoReflect()
 }
+
+func IsPLPGSQLObject(parseTree *pg_query.ParseResult) bool {
+	// CREATE FUNCTION is same parser NODE for FUNCTION/PROCEDURE
+	_, isPlPgSQLObject := parseTree.Stmts[0].Stmt.Node.(*pg_query.Node_CreateFunctionStmt)
+	return isPlPgSQLObject
+}
+
+func GetObjectTypeAndObjectName(parseTree *pg_query.ParseResult) (string, string) {
+	createFuncNode, isCreateFunc := parseTree.Stmts[0].Stmt.Node.(*pg_query.Node_CreateFunctionStmt)
```

Comment:
If this is a generic `GetObjectTypeAndObjectName` function for any input of `pg_query.ParseResult`, you probably should write this differently. 

Maybe a switch (type of parseTree.Stmts[0].Stmt.Node) and handle it for function now?

Alternatively, you could just define `GetObjectTypeAndObjectNameForFunction` and take in  input `pg_query.Node_CreateFunctionStmt`

Reply:
I kept it generic for any input based on type of stmt, I will be adding a few more to this in the second PR #1911